lib/rex/proto/mssql/client.rb

```diff
@@ -71,6 +71,57 @@ def initialize(framework_module, framework, rhost, rport = 1433, proxies = nil)
           @current_database = ''
         end
 
+        # MS SQL Server only supports Windows and Linux
+        def map_compile_os_to_platform(server_info)
+          return Msf::Platform::Unknown.realname if server_info.blank?
+
+          os_data = server_info.downcase.encode(::Encoding::BINARY)
+
+          if os_data.match?('linux')
+            platform = Msf::Platform::Linux
+          elsif os_data.match?('windows')
+            platform = Msf::Platform::Windows
+          elsif os_data.match?('win')
+            platform = Msf::Platform::Windows
+          else
+            platform = Msf::Platform::Unknown
+          end
+          platform.realname
+        end
+
+        # MS SQL Server currently only supports 64 bit but older installs may be x86
+        def map_compile_arch_to_architecture(server_info)
+          return '' if server_info.blank?
+
+          arch_data = server_info.downcase.encode(::Encoding::BINARY)
+
+          if arch_data.match?('x64')
+            arch = ARCH_X86_64
+          elsif arch_data.match?('x86')
+            arch = ARCH_X86
+          elsif arch_data.match?('64')
+            arch = ARCH_X86_64
+          elsif arch_data.match?('32-bit')
+            arch = ARCH_X86
+          else
+            arch = ''
```

Comment:
I think instead of `unknown` we'll just default to the string that the server gives back. We'll need to update the mysql logic here for consistency